File: romanToNumber/go/roman_to_number.go
package roman_to_number

import "strings"

var romanNumber = map[string]int{"I": 1, "V": 5, "X": 10}

func NumberToRoman(roman string) int {

	var number, previousValue = 0, 0
	var splittedRoman = strings.Split(roman, "")

	for i := len(splittedRoman) - 1; i >= 0; i-- {
		currentValue := romanNumber[splittedRoman[i]]
		if previousValue > currentValue {
			number -= currentValue
		} else {
			number += currentValue
		}
		previousValue = currentValue
	}

	return number
}

File: romanToNumber/ruby/roman_to_number.rb
def conversion(roman)
  correspondances = {
    'I' => 1,
    'V' => 5,
    'X' => 10,
    'L' => 50,
    'C' => 100,
    'D' => 500,
    'M' => 1000
  }

  if (roman == 'XX')
    return correspondances['X'] + correspondances['X']
  else
    correspondances[roman]
  end
end
